### ISSUE TITLE
Tutorial plans / Todo list

### ISSUE THREAD

**Issue by TwistedTwigleg** [TwistedTwigleg/Godot_FPS_Tutorial#6]
**Please remember that I am doing this in my spare time. I will chose to add, remove, change, or not work on anything listed here solely at my discretion.**

Currently this is the easiest way for me to show what is happening with this tutorial publicly. All future plans will be here for the time being. This issue will be changed as things are implemented/suggested/removed/etc.

This issue is split up into three lists:

* Upcoming plans
* Potential updates
* Changes I will not be adding

#### If you have a feature/addition request, please make a new issue for it.
Please check to see if your suggestion has already been suggested before making a new issue, to avoid duplicate requests.

____

### Quick note: I will be solely responsible for changes to this repository. See reason below:

Because this is for a tutorial, all the code has to be added (and explained) in the tutorial series. If others are adding to the repository, whether it be bug fixes, features, or additions, the tutorial in the documentation has to be changed to reflect these changes, however small they may be.

If you would like to add a feature/addition/bug-fix, please fork this repository and then add whatever you want. If you want me to (potentially) add your work into this repository, please contact me and I will look at your repository. I cannot guarantee I will add anyone else's work, but I will at very least look at it!

**Do not make pull requests on this repository. They will not be merged!**

~~Anyone who helps will be added to CONTRIBUTORS.md. Their Github user name will be listed along with what pull requests their work was included in.
I will **NOT** take credit for work done by others. **If at any point someone's work is not properly attributed to them, please let me know immediately!** I will do my very best to assure everyone gets the credit they deserve!~~

Because anyone can change the code both on the Godot documentation repository, and here on this repository, I decided to drop using CONTRIBUTERS.md. Keeping the list of contributors from both repositories in sync, and making sure I did not forget anyone between both repositories would be quite a bit of work, and thankfully Github tracks who's contributed anyway (through pull requests and issues).

**I will not take any credit for anyone's work!** I am just dropping CONTRIBUTERS.md to make things a little easier moving forward. Thanks to everyone who's helped!

#### TLDR: Do not make a pull request, but rather fork the repo, add whatever you want, and then let me know about it, if you want me to (potentially) add it. ~~If your work is used, you will be credited in CONTRIBUTORS.md.~~ If your work is used, I will not take credit for it!

____

## Upcoming plans:

None for now! :)
____

## Potential updates:

None for now! :)
____

## Changes I will not be adding:

- Multiplayer support.
- AI players

See each issue for the reasoning behind them not being added, if there is an issue attached to it.

If your request is listed here, please know it is nothing personal! :heart:

Please do not let me deter you from working on any of these if you want to! I just do not plan on working on them, nor adding them into this tutorial series!

____

## Completed updates

- Refactoring the code into more modular and performance friendly parts
- Additional input methods (controller support, changing weapons with the mouse wheel)
- Targets (Think archery targets. Likely will break into multiple pieces when destroyed)
- Turrets
- Ammo and health pickups
- Grenades
- A spawning system for players
- A sound manager (With support for background music)
- A pause menu (potentially) (Suggested in #4 )
- A main menu (potentially) (With customization settings. Will include a debug menu, as suggested in #4)
- Ability to grab and throw physics objects (potentially)
- Rebuild test level so it uses a grid map
- Make another level (using a grid map)
